groups/infrastructure/main.tf

```diff
@@ -0,0 +1,51 @@
+terraform {
+  required_version = "~> 1.3"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.54.0"
+    }
+    vault = {
+      source  = "hashicorp/vault"
+      version = "~> 3.18.0"
+    }
+  }
+}
```

Comment:
The versions are not defined as per our best practice. We should use defined lower and upper bounds.